store/idp.go

```diff
@@ -0,0 +1,268 @@
+}
+
+// ListIdentityProvider lists identity providers.
+func (s *Store) ListIdentityProvider(ctx context.Context, find *FindIdentityProviderMessage) ([]*IdentityProviderMessage, error) {
+	tx, err := s.db.BeginTx(ctx, &sql.TxOptions{ReadOnly: true})
+	if err != nil {
+		return nil, FormatError(err)
+	}
+	defer tx.Rollback()
+
+	identityProviders, err := s.listIdentityProviderImpl(ctx, tx, find)
+	if err != nil {
+		return nil, err
+	}
+	if err := tx.Commit(); err != nil {
+		return nil, FormatError(err)
+	}
+
+	return identityProviders, nil
+}
+
+// UpdateIdentityProvider updates an identity provider.
+func (s *Store) UpdateIdentityProvider(ctx context.Context, patch *UpdateIdentityProviderMessage) (*IdentityProviderMessage, error) {
+	tx, err := s.db.BeginTx(ctx, nil)
+	if err != nil {
+		return nil, FormatError(err)
+	}
+	defer tx.Rollback()
+
+	identityProvider, err := s.updateIdentityProviderImpl(ctx, tx, patch)
+	if err != nil {
+		return nil, err
+	}
+	if err := tx.Commit(); err != nil {
+		return nil, FormatError(err)
+	}
+
+	return identityProvider, nil
+}
+
+func (*Store) updateIdentityProviderImpl(ctx context.Context, tx *Tx, patch *UpdateIdentityProviderMessage) (*IdentityProviderMessage, error) {
+	set, args := []string{"updater_id = $1"}, []interface{}{fmt.Sprintf("%d", patch.UpdaterID)}
+	if v := patch.Title; v != nil {
+		set, args = append(set, fmt.Sprintf("name = $%d", len(args)+1)), append(args, *v)
+	}
+	if v := patch.Domain; v != nil {
+		set, args = append(set, fmt.Sprintf("domain = $%d", len(args)+1)), append(args, *v)
+	}
+	if v := patch.Config; v != nil {
+		set, args = append(set, fmt.Sprintf("config = $%d", len(args)+1)), append(args, *v)
+	}
+	if v := patch.Delete; v != nil {
+		rowStatus := api.Normal
+		if *patch.Delete {
+			rowStatus = api.Archived
+		}
+		set, args = append(set, fmt.Sprintf(`"row_status" = $%d`, len(args)+1)), append(args, rowStatus)
+	}
+	args = append(args, patch.ResourceID)
+
+	identityProvider := &IdentityProviderMessage{}
+	var rowStatus string
+	if err := tx.QueryRowContext(ctx, fmt.Sprintf(`
+		UPDATE idp
+		SET `+strings.Join(set, ", ")+`
+		WHERE resource_id = $%d
+		RETURNING
+			id,
+			resource_id,
+			name,
+			domain,
+			type,
+			config,
+			row_status
+	`, len(args)),
+		args...,
+	).Scan(
+		&identityProvider.UID,
+		&identityProvider.ResourceID,
+		&identityProvider.Title,
+		&identityProvider.Domain,
+		&identityProvider.Type,
+		&identityProvider.Config,
+		&rowStatus,
+	); err != nil {
+		if err == sql.ErrNoRows {
+			return nil, &common.Error{Code: common.NotFound, Err: errors.Errorf("identity provider ID not found: %s", patch.ResourceID)}
+		}
+		return nil, FormatError(err)
+	}
+
+	identityProvider.Deleted = convertRowStatusToDeleted(rowStatus)
+	return identityProvider, nil
+}
+
+func (*Store) listIdentityProviderImpl(ctx context.Context, tx *Tx, find *FindIdentityProviderMessage) ([]*IdentityProviderMessage, error) {
+	where, args := []string{"1 = 1"}, []interface{}{}
```

Comment:
```suggestion
	where, args := []string{"TRUE"}, []interface{}{}
```

This would look less hacky 😁 and PostgreSQL won't try to do the evaluation of `1 = 1`

Reply:
Maybe these changes should be separated into other PR as we are using "1=1" everywhere else?

Reply:
Sounds good!